meilisearch-types/src/locales.rs

```diff
@@ -0,0 +1,157 @@
+use deserr::Deserr;
+use milli::LocalizedAttributesRule;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+
+/// Generate a Locale enum and its From and Into implementations for milli::tokenizer::Language.
+///
+/// this enum implements `Deserr` in order to be used in the API.
+macro_rules! make_locale {
+
+    ($($language:tt), +) => {
+        #[derive(Debug, Copy, Clone, PartialEq, Eq, Deserr, Serialize, Deserialize, Ord, PartialOrd)]
+        #[deserr(rename_all = camelCase)]
+        #[serde(rename_all = "camelCase")]
+        pub enum Locale {
+            $($language),+,
+        }
+
+        impl From<milli::tokenizer::Language> for Locale {
+            fn from(other: milli::tokenizer::Language) -> Locale {
+                match other {
+                    $(milli::tokenizer::Language::$language => Locale::$language), +
+                }
+            }
+        }
+
+        impl From<Locale> for milli::tokenizer::Language {
+            fn from(other: Locale) -> milli::tokenizer::Language {
+                match other {
+                    $(Locale::$language => milli::tokenizer::Language::$language), +,
+                }
+            }
+        }
+
+        #[derive(Debug)]
+        pub struct LocaleFormatError {
+            pub invalid_locale: String,
+        }
+
+        impl std::fmt::Display for LocaleFormatError {
+            fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                let valid_locales = [$(Locale::$language),+].iter().map(|l| format!("`{}`", json!(l).as_str().unwrap())).collect::<Vec<_>>().join(", ");
+                write!(f, "Unknown value `{}`, expected one of {}", self.invalid_locale, valid_locales)
```

Comment:
Should we change the phrasing here a little bit?

```suggestion
                write!(f, "Unsupported locale `{}`, expected one of {}", self.invalid_locale, valid_locales)
```

?